src/main/java/fi/aalto/cs/apluscourses/presentation/MainViewModel.java

```diff
@@ -121,6 +127,9 @@ public void readAuthenticationFromStorage(@Nullable PasswordStorage passwordStor
         .map(PasswordStorage::restorePassword)
         .map(factory::create)
         .ifPresent(this::setAuthentication);
+    if (!exercisesViewModel.get().isAuthenticated()) {
+      this.updateExercises();
+    }
```

Comment:
I'm not sure if it makes sense to ensure `updateExercises()` is called here. That should perhaps be done by the caller.

Reply:
BTW what's the reason that the call is needed here in the first place?

Reply:
There is an earlier comment in this PR that sheds some light. I am not 100% sure about this but I think the view was not updated in a case where there was no token set and I needed to manually call updateExercises. I tested that everything works without it now so I can remove it.